scipy/spatial/transform/_rotation.pyx

```diff
@@ -1865,6 +1873,118 @@ cdef class Rotation:
+        applied to the angle of rotation about the rotation's fixed axis. The
+        expression ``q = p ** n`` can also be expressed as
+        ``q = Rotation.from_rotvec(n * p.as_rotvec())``
+
+        Parameters
+        ----------
+        n : `double`
+            The number of times to compose the rotation with itself. ``n`` must
+            be positive.
+        modulus : None
+            This overridden argument is not applicable to Rotations and must be
+            None.
+
+        Returns
+        -------
+        composition : `Rotation` instance
+            If the input Rotation ``p`` contains ``N`` multiple rotations, then
+            the output will contain ``N`` rotations where the ``i``th rotation
+            is equal to ``p[i] ** n``
+
+        Notes
+        -----
+        For example, a power of 2 will double the angle of rotation, and a
+        power of 0.5 will halve the angle. There are two notable cases: if
+        ``n == 1`` then the original rotation is returned, and if ``n == 0``
+        then the identity rotation is returned.
+
+        Note that fractional powers ``0 < n < 1`` which effectively take a root
+        of rotation, do so using the shortest path smallest representation of
+        that angle (the principal root). This means that powers of ``n`` and
+        ``1/n`` are not necessarily inverses of each other. For example, a 0.5
+        power of a +240 degree rotation will be calculated as the 0.5 power of
+        a -120 degree rotation, with the result being a rotation of -60 rather
+        than +120 degrees.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Raising a rotation to a power:
+
+        >>> p = R.from_rotvec([1, 0, 0])
+        >>> q = p ** 2
+        >>> q.as_rotvec()
+        array([2., 0., 0.])
+        >>> r = p ** 0.5
+        >>> r.as_rotvec()
+        array([0.5, 0., 0.])
+
+        Inverse powers do not necessarily cancel out:
+
+        >>> p = R.from_rotvec([0, 0, 120], degrees=True)
+        >>> ((p ** 2) ** 0.5).as_rotvec(degrees=True)
+        array([  -0.,   -0., -60.])
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double angle, scale, angle2, n3, n5
+        cdef double[:, :] result = _empty2(num_rotations, 4)
+        cdef double[:] quat
+
+        if modulus is not None:
+            raise NotImplementedError("modulus not supported")
+
+        if n < 0:
+            raise ValueError("Rotations cannot be raised to negative powers.")
+
+        # short-cuts
+        elif n == 0:
+            return Rotation.identity(num_rotations)
+
+        elif n == 1:
+            result = self._quat.copy()
+
+        elif n == 2:
+            result = _compose_quat(self._quat, self._quat)
+
+        # general scaling of rotation angle
+        else:
+            for ind in range(num_rotations):
+                if self._quat[ind, 3] < 0:  # w > 0 to ensure 0 <= angle <= pi
```

Comment:
Since this is not performed for the cases `n == 2` and `n == 1`, wouldn't this cause some minor inconsistent behavior?

Reply:
There will be some slightly inconsistent behavior for the n=2 case, for a 90 deg rotation that gets squared to the opposite pole. See some discussion [here](https://github.com/scipy/scipy/pull/17334) about that case (aka the q[3] == 0 case). I think it’s not so much an issue for n=2 since it’s a marginal epsilon scenario, as it might result in different representations for n=2 and n=5 (which for a 90 deg rotation should be the same).

Reply:
I see what you mean! 
Personally I'd prefer to do the same to all of them for consistency, specially since `n=2` could be possible the most used case.

Reply:
Just a last comment on this, I've quickly read through your PR about mapping the quaternion to a single cover.

I'd argue that the fact it hasn't been decided (at least for now) whether to merge it or not is a good argument to remove this from here, and let the decision be made elsewhere in the code, as a parameter, etc.

Even if the single cover mapping does get merged, but then someday in the future for some reason it gets reversed, it would probably be better not to have this kind of leftover behavior.

What do you think?

Reply:
The single cover mapping here is a quick way to take the shortest angle path from one point to another, rather than accidentally taking the opposite direction `2*pi - angle` path. The same logic is in `as_rotvec()`. It’s necessary for proper functioning, and unrelated to the single cover MR which is about how to represent things.

Reply:
It will be inconsistent in the double cover quaternion space, yes. It will be the same in every other representation. In general I don’t think we should rely on the properties of the double cover space since the abstract object here is a Rotation rather than a quaternion, and users might only be familiar with one of the representations (ie rotation matrices). If we weren’t to enforce the shortest path difference then we would have a well defined operation on the quaternions, but we would force users into using quaternions. This could result in unexpected behavior if mixing representations. I’m all ears though if there’s a clever way to mitigate this.

Reply:
I do think it would be a good idea to scope out some documentation and tests to define the explicit scope of where the double cover properties are retained. Likely just `__mul__`, `.inv()`, `.from_quat()`, and `.as_quat(canonical=False)`. Probably a separate PR though.

Reply:
I see what you mean, yes! A agree that the choice should be made focused on how we "see" it as an abstract "Rotation" and not as a quaternion. What I mean though is that if this gets accepted, maybe some thought should be given in whether it would be beneficial to also apply this to the simple quaternion composition.

Reply:
Maybe, I think it warrants some more discussion.

Reply:
Personally I would prefer to leave both things completely separated. 

I'd define __pow__ as the direct quaternion power, for consistency with __add__ being equivalent to a direct quaternion composition. Mostly, because I think this inconsistency might also create bugs that are hard to analyze for someone not familiar with how they are defined.

Then maybe in the class initialiser we could have a final check that always assures the stored quaternion has positive single part (except if a special flag is set to False, as you said!)

Reply:
I haven't taken out the `n == 2` case here yet but once I do I'm not sure I'm following what your recommended change is?

Reply:
Sorry if I'm mistaken, I'm currently seeing this from my phone. Here's what I mean:

Currently, the code first maps the quaternion to the canonical cover, then calculates the angle, then multiplies the angle by n.

This multiplied angle can be bigger than pi though, and no more mapping is made after that, so the resulting quaternion can be possibly on the non canonical cover. Am I right?

What I propose is to first remove the check before multiplying the angle, and then, let all the mapping happen at the very end, when calling `__class__`.

In this case, the class initialiser would deal with the cover mapping.

Again sorry if I'm mistaken, hopefully I managed to be clearer!

Reply:
I think I see what you mean! But I think this may run into issues with fractional powers. Let me play with it when I get home.

Reply:
Removed the `n==2` case and added support for negative exponents.